plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/project/interceptor/AbstractJavaContentRootInterceptor.java

```diff
@@ -88,30 +88,17 @@ private ContentRoot getSourceType(Node node) {
     }
 
     private String _getSourceFolder(ProjectConfigDto projectConfig, String srcAttribute) {
-        Map<String, List<String>> attributes = projectConfig.getAttributes();
+        final Map<String, List<String>> attributes = projectConfig.getAttributes();
         if (!attributes.containsKey(srcAttribute)) {
-            return "";
+            return null;
```

Comment:
Because maven project (which is java simultaneously) should have source folder definition, in this case it will be defined or empty, if empty, then we'll mark root folder source. If project is java (not maven), then source folder may not be defined, if so, then we don't mark any folder as source.